selfdrive/statsd.py

```diff
@@ -104,8 +116,22 @@ def get_influxdb_line(measurement: str, value: float, timestamp: datetime, tags:
       for gauge_key in gauges:
         result += get_influxdb_line(f"gauge.{gauge_key}", gauges[gauge_key], current_time, tags)
 
+      for sample_key in samples:
+        samples[sample_key].sort()
+        sample_count = len(samples[sample_key])
+        sample_sum = sum(samples[sample_key])
+
+        result += get_influxdb_line(f"sample.{sample_key}.count", sample_count, current_time, tags)
+        result += get_influxdb_line(f"sample.{sample_key}.min", samples[sample_key][0], current_time, tags)
+        result += get_influxdb_line(f"sample.{sample_key}.max", samples[sample_key][-1], current_time, tags)
+        result += get_influxdb_line(f"sample.{sample_key}.mean", sample_sum / sample_count, current_time, tags)
+        for percentile in [0.05, 0.5, 0.95]:
+          value = samples[sample_key][int(round(percentile * (sample_count - 1)))]
+          result += get_influxdb_line(f"sample.{sample_key}.p{int(percentile * 100)}", value, current_time, tags)
```

Comment:
```suggestion
      for key, values in samples.items():
        values.sort()
        sample_count = len(values)
        sample_sum = sum(values)

        result += get_influxdb_line(f"sample.{key}.count", sample_count, current_time, tags)
        result += get_influxdb_line(f"sample.{key}.min", values[0], current_time, tags)
        result += get_influxdb_line(f"sample.{key}.max", values[-1], current_time, tags)
        result += get_influxdb_line(f"sample.{key}.mean", sample_sum / sample_count, current_time, tags)
        for percentile in (0.05, 0.5, 0.95):
          value = values[int(round(percentile * (sample_count - 1)))]
          result += get_influxdb_line(f"sample.{sample_key}.p{int(percentile * 100)}", value, current_time, tags)
```

Reply:
I feel like these should be fields on a single metric instead of separate metrics?

Reply:
that does indeed sound better, did it this way since statsd seems to also make multiple entries (not exactly clear why).

Reply:
tags/fields are not features of vanilla statsd